dist/XSLoader/t/XSLoader.t

```diff
@@ -21,6 +16,102 @@ BEGIN {
     }
 }
 
+# mini test implementation. We're going to be playing with the XS bits of
+# various modules that may be used by Test::More, so it's best to avoid. Since
+# XSLoader is dual life, we can't use something like perl's t/test.pl
+my $planned_tests;
+my $tests;
+my $passed_tests;
+sub ok ($;$) {
+    my ($ok, $name) = @_;
+    $tests++;
+    $passed_tests += 1 if $ok;
+    print STDOUT "not "
+        if !$ok;
+    print STDOUT "ok $tests";
+    print STDOUT " - $name"
+        if defined $name;
+    print "\n";
+    return $ok;
+}
+sub is ($$;$) {
+    my ($got, $want, $name) = @_;
+
+    my $ok
+        = !defined $want && !defined $got
+        || defined $want && defined $got && $got eq $want;
+
+    defined $_ or $_ = '[undef]'
+        for $got, $want;
+
+    ok($ok, $name)
+        or diag("Got: $got\nExpected: $want");
+
+    return $ok;
+}
+sub can_ok ($@) {
+    my ($inv, @methods) = @_;
+    die "only supports one method"
+        if @methods != 1;
+    ok $inv->can($methods[0]), "$inv->can('$methods[0]')";
+}
+sub skip ($$) {
+    my ($message, $tests) = @_;
+    die "bad skip"
+        if !$tests || $tests =~ /[^0-9]/;
+    $tests += $tests;
+    $passed_tests += $tests;
+    no warnings 'exiting';
+    last SKIP;
+}
+sub like ($$;$) {
+    my ($got, $want_re, $name) = @_;
+    if (!ref $want_re) {
+        $want_re =~ m{\A/(.*)/([a-z]*)\z}
+            or die "bad regex $want_re";
+        $want_re = (length $2 ? "(?$2)" : '') . $1;
+    }
+    my $ok = $got =~ $want_re;
+    ok($ok, $name)
+        or diag("Got: $got\nExpected: $want_re");
+    return $ok;
+}
+sub diag {
+    my ($message) = @_;
+    $message =~ s/\n?\z/\n/;
+    $message =~ s/^/# /gm;
+    print STDERR $message;
+}
+END {
+    if (!defined $planned_tests) {
+        $planned_tests = $tests;
+        print STDOUT "1..$planned_tests\n";
+    }
```

Comment:
This is a little dangerous. If the file is changed to not use a plan, which appears to be the intent of this code, this would allow the file to pass even if one of the tests in it does an early exit.

I think this should die if there's no plan set.

If we want to support no-plan type testing, the code above should die and we add:
```
sub done_testing {
    $planned_tests = $tests;
}
```
which ensures execution reaches the call to done_testing() and behaves like no-plan testing in modern Test::More.

Reply:
I'd agree that no_plan style testing is dangerous. Since this test doesn't need it, I've just removed this and replaced it with an error.